Make sync-frontend-brand-assets missing-asset test independent from /tmp sibling repos

## Summary
The full Android test suite currently fails in `tests/sync-frontend-brand-assets.test.ts` because the negative-path assertion depends on `/tmp/frontend/public/...` not existing on the host machine.

## Evidence
Local run on 2026-04-24:

```text
npm run test:run
```

Failure:

```text
FAIL  tests/sync-frontend-brand-assets.test.ts > frontend brand asset sync > fails fast when the canonical frontend logo assets are unavailable
AssertionError: expected [Function] to throw an error
```

The test calls:

```ts
buildFrontendBrandAssetPlan("/tmp/brand-sync-missing-assets")
```

The production helper resolves sibling frontend assets via `resolve(repoRoot, "../frontend/public")`, so if `/tmp/frontend/public/logo-source.png` exists for unrelated reasons, the negative-path assertion becomes false and the test stops being deterministic.

## Expected direction
Make the negative-path test use a guaranteed unique temp root or otherwise isolate file-existence checks so the assertion does not depend on ambient files outside the repository.

## Scope note
This was discovered while validating the separate `@xmldom/xmldom` security update and is intentionally tracked outside that PR to preserve one-topic scope.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -26,6 +26,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- `tests/sync-frontend-brand-assets.test.ts` now uses an isolated temporary repo root for its missing-asset assertion, so the Android suite no longer depends on unrelated `/tmp/frontend` leftovers on the host machine
 - bumped the repo-local `@xmldom/xmldom` npm override to `0.8.13`, clearing the high-severity processing-instruction XML injection advisory and the related xmldom audit findings from the Android Capacitor CLI dependency chain
 - Android passkey auth now maps Credential Manager unsupported/provider failures via explicit AndroidX exception types instead of class-name heuristics, so unsupported-device/provider states consistently surface the native `PASSKEY_PROVIDER_UNAVAILABLE` path used by the shared login UI.
 - The Android wrapper now declares `asset_statements` for `https://app.secpal.dev/.well-known/assetlinks.json` in its manifest resources, aligning the installed app with Android Credential Manager's Digital Asset Links prerequisite for passkey RP-ID validation.

--- a/tests/sync-frontend-brand-assets.test.ts
+++ b/tests/sync-frontend-brand-assets.test.ts
@@ -3,6 +3,9 @@
  * SPDX-License-Identifier: AGPL-3.0-or-later
  */
 
+import { mkdirSync, mkdtempSync, rmSync } from "node:fs";
+import { tmpdir } from "node:os";
+import { join, resolve } from "node:path";
 import { describe, expect, it } from "vitest";
 
 async function loadBrandSyncModule(): Promise<{
@@ -38,14 +41,25 @@ describe("frontend brand asset sync", () => {
       assertFrontendBrandAssetSourcesExist,
       buildFrontendBrandAssetPlan,
     } = await loadBrandSyncModule();
+    const tempRoot = mkdtempSync(join(tmpdir(), "brand-sync-missing-assets-"));
+    const isolatedRepoRoot = resolve(tempRoot, "android");
 
-    expect(() =>
-      assertFrontendBrandAssetSourcesExist(
-        buildFrontendBrandAssetPlan("/tmp/brand-sync-missing-assets")
-      )
-    ).toThrowError(
-      "Missing canonical frontend brand asset: /tmp/frontend/public/logo-source.png"
-    );
+    mkdirSync(isolatedRepoRoot, { recursive: true });
+
+    try {
+      expect(() =>
+        assertFrontendBrandAssetSourcesExist(
+          buildFrontendBrandAssetPlan(isolatedRepoRoot)
+        )
+      ).toThrowError(
+        `Missing canonical frontend brand asset: ${resolve(
+          tempRoot,
+          "frontend/public/logo-source.png"
+        )}`
+      );
+    } finally {
+      rmSync(tempRoot, { recursive: true, force: true });
+    }
   });
 
   it("maps canonical frontend logos to Android launcher and splash outputs", async () => {

--- a/tests/sync-frontend-brand-assets.test.ts
+++ b/tests/sync-frontend-brand-assets.test.ts
@@ -44,9 +44,9 @@ describe("frontend brand asset sync", () => {
     const tempRoot = mkdtempSync(join(tmpdir(), "brand-sync-missing-assets-"));
     const isolatedRepoRoot = resolve(tempRoot, "android");
 
-    mkdirSync(isolatedRepoRoot, { recursive: true });
-
     try {
+      mkdirSync(isolatedRepoRoot, { recursive: true });
+
       expect(() =>
         assertFrontendBrandAssetSourcesExist(
           buildFrontendBrandAssetPlan(isolatedRepoRoot)

